### ISSUE TITLE
Game.ini settings on an instance specific base

### ISSUE THREAD

**Issue by maxbrand99** [FezVrasta/ark-server-tools#1035]
I am trying to change the settings of only one of the servers that I am running. I want to disable friendly fire on one of my maps in the cluster. I read somewhere here about duplicating the entire ark server files. Is there any way to change it without doing this? I do not have the space to duplicate ark on my system.

**Comment by klightspeed** [comment]
If you don't want to use a staging directory, then you can try using bind mounts:
```
mount --bind /path/to/common/ark/directory /path/to/instance/directory
mount --bind /path/to/instance/save /path/to/instance/directory/ShooterGame/Saved
mount --bind /path/to/instance/mods /path/to/instance/directory/ShooterGame/Content/Mods
```

**Comment by maxbrand99** [comment]
How would I use this to have different game.ini files?

**Comment by klightspeed** [comment]
Using bind mounts would allow you to use the same game files while keeping the saved files (including Game.ini and GameUserSettings.ini) separate between instances.

**Comment by maxbrand99** [comment]
another question then as i am not super fluent in linux yet, how would i do that? just by copying and pasting the command above?
That seems so complicated, is ASM not able to edit Game.ini like it can for GameUserSettings.ini? I know for GameUserSettings.ini i can do ark_ arkflag_ or arkopt_ (i add all 3 but dont actually know which one works) for changing settings based on the instance but it does not work for lines in Game.ini

**Comment by klightspeed** [comment]
In the mount commands I gave, replace:
* `/path/to/common/ark/directory` with the path to the common ark directory
* `/path/to/instance/save` with the path to the directory where the instance save files should be stored
* `/path/to/instance/mods` with the path to the directory where the instance mods should be stored
* `/path/to/instance/directory` with the path to the directory from which you want the instance to run.

arkmanager does not itself modify GameUserSettings.ini - rather arkmanager provides the command-line options, and the game server uses these command-line options to modify the config and then saves the modified config.

ark_optionname=value options are those specified in the option string `mapname?option1=value1?option2=value2`
These are the `?optionname=value` options specified in <https://ark.gamepedia.com/Server_Configuration#Command_line_arguments>

arkopt_optionname=value options are those specified with a leading dash with a value
These are the `-optionname=value` options specified in <https://ark.gamepedia.com/Server_Configuration#Command_line_arguments>

arkflag_optionname=true options are those specified with a leading dash without a value
These are the `-optionname` options specified in <https://ark.gamepedia.com/Server_Configuration#Command_line_arguments>

**Comment by Partyznat** [comment]
Can you tell me what I'm doing wrong that I don't want the test instance to work?
mount --bind /home/arkserver/ARK /home/arkserver/instance/ARK.center
mount -- bind /home/arkserver/instance/ARK.center.mods /home/arkserver/ARK/ShooterGame/Content/Mods
mount --bind /arkserver/instance/ARK.center.save /home/arkserver/ARK/ShooterGame/Saved

And I changed the root instance /home/arkserver/instance/ARK.center 
IN Main.cfg

**Comment by klightspeed** [comment]
It looks like you meant to:
```
mount --bind /home/arkserver/ARK /home/arkserver/instance/ARK.center
mount -- bind /home/arkserver/instance/ARK.center.mods /home/arkserver/ARK.center/ShooterGame/Content/Mods
mount --bind /arkserver/instance/ARK.center.save /home/arkserver/ARK.center/ShooterGame/Saved
```

**Comment by Partyznat** [comment]
Thank you I changed but still the same problem does not start stops at 2.34 GB of RAM consumption

**Comment by Partyznat** [comment]
Yes, each instance is assigned a different one.  As I use on these given folders umount -A works fine

**Comment by Partyznat** [comment]
S_API FAIL] SteamAPI_Init() failed; SteamAPI_IsSteamRunning() failed.

**Comment by klightspeed** [comment]
> S_API FAIL] SteamAPI_Init() failed; SteamAPI_IsSteamRunning() failed.

This is normal - it just means that you don't have the Steam client running, which is normal on a server.

**Comment by Partyznat** [comment]
I'm an idiot ... It was enough to restart the whole server to start everything working.  Thanks for the help

**Comment by Soldize** [comment]
Hi, 

i put the mount commands in "/etc/arkmanager/instances/servername.cfg" right??

best regards

**Comment by klightspeed** [comment]
No.

If you want to go with persistent bind mounts, then the best place to put
the mounts is in /etc/fstab.

/home/arkserver/ARK /home/arkserver/instance/ARK.center bind defaults 0 0

/home/arkserver/instance/ARK.center.mods
/home/arkserver/ARK.center/ShooterGame/Content/Mods bind defaults 0 0

/home/arkserver/instance/ARK.center.save
/home/arkserver/ARK.center/ShooterGame/Saved bind defaults 0 0


To mount these without restarting, run the following command:

mount -a
